File: Medium/735. Asteroid Collision/Solution.py
class Solution:
    def asteroidCollision(self, asteroids: list[int]) -> list[int]:
        stack = []
        
        for ast in asteroids:
            yasiyorMu = True
            
            while stack and stack[-1] > 0 and ast < 0:
                if stack[-1] < abs(ast):
                    stack.pop()
                    continue
                elif stack[-1] == abs(ast):
                    stack.pop()
                
                yasiyorMu = False
                break
            
            if yasiyorMu:
                stack.append(ast)
        
        return stack

File: Medium/735. Asteroid Collision/Solution.java
import java.util.Stack;

class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> stack = new Stack<>();

        for (int ast : asteroids) {
            boolean yasiyorMu = true;

            while (!stack.isEmpty() && stack.peek() > 0 && ast < 0) {
                if (stack.peek() < Math.abs(ast)) {
                    stack.pop();
                    continue; 
                } else if (stack.peek() == Math.abs(ast)) {
                    stack.pop();
                }
                
                yasiyorMu = false;
                break;
            }

            if (yasiyorMu) {
                stack.push(ast);
            }
        }

        int[] result = new int[stack.size()];
        for (int i = result.length - 1; i >= 0; i--) {
            result[i] = stack.pop();
        }

        return result;
    }
}